source/extensions/filters/listener/proxy_protocol/config.cc

```diff
@@ -28,7 +28,8 @@ class ProxyProtocolConfigFactory : public Server::Configuration::NamedListenerFi
         const envoy::extensions::filters::listener::proxy_protocol::v3::ProxyProtocol&>(
         message, context.messageValidationVisitor());
 
-    ConfigSharedPtr config = std::make_shared<Config>(context.scope(), proto_config);
+    ConfigSharedPtr config =
+        std::make_shared<Config>(context.scope(), context.listenerScope(), proto_config);
```

Comment:
I like putting the new stats under the listener scope, but this is inconsistent with the previous stats, and with other listener filter stats (such as tls_inspector). @jmarantz any advice or thoughts on which scope to put listener filter stats under?

Reply:
@jmarantz context (so you don't have to read all the other discussions)

- This filter currently has a single stat `downstream_cx_proxy_proto_error` at the root scope.
- This PR introduces a few finer-grained stats that capture all the information present in `downstream_cx_proxy_proto_error`. However, I would like to put them under the listener's scope, since this is a listener filter.
- The new stats introduced here basically make `downstream_cx_proxy_proto_error` obsolete, but I will keep it for backwards compatibility.

Reply:
PTAL @jmarantz , otherwise I will assume this is OK

Reply:
I looked around, and it looks like `listenerScope()` should only be used for listener-specific stats, not for extensions. Probably the way to handle this is to add a `stat_prefix` config and use that; this is what most/all extensions do. It would be easier to review if you do that in a followup PR, and just use `context.scope()` here.

Reply:
Makes sense. Ok, removed all listener scope plumbing and keeping the stats under the root scope. We can add stat prefix in follow-up PR.